tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -365,8 +352,10 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_stuffer_alloc(&wire_identities_in, sizeof(wire_identites)));
             EXPECT_SUCCESS(s2n_stuffer_write_bytes(&wire_identities_in, wire_identites, sizeof(wire_identites)));
 
-            EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+            EXPECT_ERROR_WITH_ERRNO(s2n_client_psk_recv_identity_list(conn, &wire_identities_in),
+                                      S2N_ERR_VALID_PSK_IDENTITY_NOT_FOUND);
```

Comment:
This behavior should not change. No match should not be an error, it should be a no-op.